pkgs/os-specific/linux/systemd/default.nix

```diff
@@ -1,22 +1,57 @@
-{ stdenv, lib, fetchFromGitHub, pkgconfig, intltool, gperf, libcap
-, curl, kmod, gnupg, gnutar, xz, pam, acl, libuuid, m4, e2fsprogs, utillinux, libffi
-, glib, kbd, libxslt, coreutils, libgcrypt, libgpgerror, libidn2, libapparmor
-, audit, lz4, bzip2, pcre2
-, linuxHeaders ? stdenv.cc.libc.linuxHeaders
-, iptables, gnu-efi, bashInteractive
-, gettext, docbook_xsl, docbook_xml_dtd_42, docbook_xml_dtd_45
-, ninja, meson, python3Packages, glibcLocales
-, patchelf
-, substituteAll
-, getent
-, cryptsetup, lvm2
+{ stdenv, lib, fetchFromGitHub
 , buildPackages
-, perl
+, ninja, meson, m4, pkgconfig, coreutils, gperf, getent
+, patchelf, perl, glibcLocales, glib, substituteAll
+, gettext, python3Packages
+
+# Mandatory dependencies
+, libcap
+, utillinux
+, kbd
+, kmod
+
+# Optional dependencies
+, pam, cryptsetup, lvm2, audit, acl
```

Comment:
We could/should probably add optional dependencies for `libp11-kit` and `libfido2` .  Both can be used to unlock systemd-homed partitions using your yubikey. which is one of the cool main features of homed.

The first one provides support to unlock using PKCS11, whilst the other uses the more modern FIDO2 protocol for crypto.


However this might potentially add an interesting cycle; as `libfido2` depends on `udev`.  So we might need to build `udev` with `  { libfido2 = null; }`

Reply:
And wouldn't that be a reason to keep `systemdMinimal`?

Reply:
is `libp11-kit` in `libp11`?

Reply:
Nope it's in `nixpkgs.p11-kit`

Reply:
systemd-homework now links against libfido2 and libp11-kit. So I think it works